### PR TITLE
Dra 659 paralleliser transforationer

### PR DESCRIPTION
**Pull request by davidgrove73**

### CODE REVIEW

**Review by VictorHarbo — CHANGES_REQUESTED** [review]
Looks good overall. However, when I am running individual test classes they fail due to initialization errors. Could you please make sure that all `transformer` test classes can be run individually from the IDE. The same goes for the class `ViewTest`.  

I would also like some kind of test of the metadata map not getting messed up. Maybe you could add a test in ViewTest, with multiple test records being processed in parallel and checking the solr transformation for value: `internal_storage_mTime` as te storage value can be set directly on DsRecords with something like this ` DsRecordDto recordDto = new DsRecordDto().data(pvica).id("test.id").mTime(1701261949625000L);`

**Comment by davidgrove73** [comment]
ServiceConfig needed to be initialized before the tests are run. Not sure if this is the best way of doing it and I am a little unsure of when static variables are initialized when tests are run.

An test for testing parallel processing of records have been added to views.

**Review by VictorHarbo — CHANGES_REQUESTED** [review]
I like the tests! Small suggestion on how the conf is loaded.

File: src/test/java/dk/kb/present/solr/EmbeddedSolrTest.java [hunk 1]
```diff
@@ -76,7 +77,15 @@ public class EmbeddedSolrTest {
...
-    public static void startEmbeddedSolrServer() {
+    public static void startEmbeddedSolrServer() throws IOException {
+
+        String CONFIG = Resolver.resolveGlob("conf/ds-present-behaviour.yaml").get(0).toString();
+        if ("[]".equals(CONFIG)) {
+            throw new IllegalStateException("Unable to locate config");
+        }
+
+        log.info("Fixing config to '{}'", CONFIG);
+        ServiceConfig.initialize(CONFIG);
```
[the remark this answers is not in the document]
Comment (davidgrove73) [reply src/test/java/dk/kb/present/solr/EmbeddedSolrTest.java]: I think you are right, and ServiceConfig.initialize handles missing config files as well.

**Pull request merged**